### PR TITLE
Fix Facebook TOS & PP

### PR DESCRIPTION
**Pull request by afisher3578**
Fix blinking from issues #24 #32

### CODE REVIEW

File: declarations/Facebook.json [hunk 1]
```diff
@@ -15,12 +15,18 @@
       "executeClientScripts": true
     },
     "Privacy Policy": {
-      "fetch": "https://www.facebook.com/about/privacy",
+      "fetch": "https://mbasic.facebook.com/privacy/policy/printable/",
```
Comment (clementbiron) [inline declarations/Facebook.json:18]: Why did you change the fetch url while the document he's still available on https://www.facebook.com/privacy/policy/ ?

File: declarations/Facebook.json [hunk 2]
```diff
@@ -15,12 +15,18 @@
...
       "filter": [
-        "removeReturnToTopButtons",
-        "cleanUrls"
+        "removeEmptyAnchorsLinks",
+        "removeTrackingIDs",
+        "removeLocaleFromUrls"
       ],
-      "select": "._5tkp"
+      "select": [ 
+        "div[id=root]"
```
Comment (clementbiron) [inline declarations/Facebook.json:25]: `#root` seems simpler to me

File: declarations/Facebook.json [hunk 3]
```diff
@@ -15,12 +15,18 @@
...
+      ],
+      "remove": [
+        "div[class=k]"
```
Comment (clementbiron) [inline declarations/Facebook.json:28]:
```suggestion
        ".k"
```

Comment (clementbiron) [reply declarations/Facebook.json:28]: Images should be removed with `.l`

Comment (clementbiron) [reply declarations/Facebook.json:28]: _Return to the top_ elements should be removed with `.bd`

File: declarations/Facebook.json (hunk 2 above)
Comment (clementbiron) [reply declarations/Facebook.json:25]: And to be more consistent you could use the same selector as the ToS: `div[role=main]`

File: declarations/Facebook.json [hunk 4]
```diff
@@ -15,12 +15,18 @@
       "executeClientScripts": true
     },
     "Privacy Policy": {
-      "fetch": "https://www.facebook.com/about/privacy",
+      "fetch": "https://mbasic.facebook.com/privacy/policy/printable/",
       "filter": [
-        "removeReturnToTopButtons",
-        "cleanUrls"
+        "removeEmptyAnchorsLinks",
```
Comment (clementbiron) [inline declarations/Facebook.json:20]: `document.querySelectorAll('[href="#"]')` length is 0, so i think this filter is not necessary.

File: declarations/Facebook.json [hunk 5]
```diff
@@ -15,12 +15,18 @@
...
+        "removeTrackingIDs",
+        "removeLocaleFromUrls"
```
Comment (clementbiron) [inline declarations/Facebook.json:22]: `locale2` query param does not seem to me to be present in the urls of the document, so the filter should not be necessary

**Review by clementbiron — CHANGES_REQUESTED** [review]

File: declarations/Facebook.json (hunk 1 above)
Comment (afisher3578) [reply declarations/Facebook.json:18]: The printable version was easier to inspect and select parts to crawl, but if that is not best practice to change the url that we crawl, I can change it back

Comment (MattiSG) [reply declarations/Facebook.json:18]: Quite the contrary, I think using printable versions is a great way to simplify selection, reduce use of filters, and improve stability! 👏 We could probably try disabling `executeClientScripts` on this version 🙂

Comment (afisher3578) [reply declarations/Facebook.json:18]: @MattiSG Currently the privacy policy does not have `executeClientScripts` should I add that and set it to false? 

Or do you mean to remove `executeClientScripts` from ToS since that is a plaintext page

Comment (MattiSG) [reply declarations/Facebook.json:18]: > should I add that and set it to false?

You don't need to do that since [the default is `false`](https://github.com/OpenTermsArchive/contrib-declarations/blob/main/CONTRIBUTING.md#executeclientscripts) 🙂

> do you mean to remove `executeClientScripts` from ToS since that is a plaintext page

It is worth checking if your URL change also enables removing `executeClientScripts` from other pages, yes!

File: declarations/Facebook.json [hunk 6]
```diff
@@ -4,7 +4,7 @@
     "Terms of Service": {
       "fetch": "https://www.facebook.com/legal/terms/plain_text_terms",
       "select": [
-        ".ad2k81qe"
+        "div[role=main]"
       ],
       "remove": [
         ".mdeji52x"
```
Comment (clementbiron) [inline declarations/Facebook.json:10]: An improvement i missed last time, you could cleaning the version by adding the following remove selector:
`.bdao358l.om3e55n1.g4tp4svg.alzwoclg.cqf1kptm.jez8cy9q.gvxzyvdx.i85zmo3j.q6feio67.srn514ro`

File: declarations/Facebook.json [hunk 7]
```diff
@@ -15,12 +15,15 @@
...
+      "fetch": "https://mbasic.facebook.com/privacy/policy/printable/",
+      "select": [
+        "#root"
       ],
-      "select": "._5tkp"
+      "remove": [
+        ".k",
+        ".l",
+        ".bd"
+      ]
```
Comment (clementbiron) [inline declarations/Facebook.json:26]: I was wrong, the selector to remove the "return to the top" elements generate an empty link in the version and this is not good. It should be updated with `.bc`

Another improvement, we could remove unecessary images with `img[style=\"display:none\"]`

```suggestion
        ".bc",
        "img[style=\"display:none\"]"
      ]
```

**Review by clementbiron — CHANGES_REQUESTED** [review]

**Review by clementbiron — APPROVED** [review]
